Controllers/WishListsController.cs

```diff
@@ -0,0 +1,256 @@
+            var user = _wca.GetContext().CurrentUser;
+            if (user == null) {
+                return new HttpUnauthorizedResult();
+            }
+            //get selected wishlist
+            var wishList = _wishListServices.GetWishList(user, wishListid);
+
+            if (wishList.ContentItem.Id == wishListid) { //GetWIshList may return the default wish list
+                _wishListServices.RemoveElementFromWishlist(user, wishList, elementId);
+            }
+
+            return RedirectToAction("Index", new { id = wishListid });
+        }
+
+        [HttpPost]
+        public ActionResult UpdateSettings(int wishListid) {
+            var user = _wca.GetContext().CurrentUser;
+            if (user == null) {
+                return new HttpUnauthorizedResult();
+            }
+            var wishLists = _wishListServices.GetWishLists(user);
+            //1. Read what should change from the form
+            var form = HttpContext.Request.Form;
+            //1.1 Get the id of the new default wish list
+            var defaultId = int.Parse(form["is-default-wishlist"]);
+            //1.2 Get the new titles
+            var newTitles = form.AllKeys
+                .Where(key => key.StartsWith("wishlist-title-"))
+                .ToDictionary(
+                    key => int.Parse(key.Substring("wishlist-title-".Length)),
+                    key => form[key]
+                );
+            //1.3 Get the list of wish lists to delete
+            var toDelete = new List<int>();
+            if (form.AllKeys.Contains("delete-wishlist")) {
+                toDelete.AddRange(form["delete-wishlist"]
+                    .Split(new string[] { "," }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(int.Parse));
+            }
+            foreach (var wishList in wishLists) {
+                var wlId = wishList.ContentItem.Id;
+                if (toDelete.Contains(wlId)) {
+                    //Delete this list
+                    _wishListServices.DeleteWishlist(user, wishList);
+                }
+                else {
+                    //2. Update titles
+                    var title = newTitles[wlId];
+                    wishList.ContentItem.As<TitlePart>().Title = title;
+                    //3. Update default wishlist
+                    wishList.IsDefault = wlId == defaultId;
+                    //4. Process extension behaviours
+                    foreach (var ext in _wishListExtensionProviders) {
+                        ext.UpdateSettings(user, wishList);
+                    }
+                }
+            }
+            //we may have deleted the default wishlist
+            //that condition is handled in the GetDefaultWishList method
+
+            return RedirectToAction("Index", new { id = wishListid });
+        }
+
+        private Dictionary<int, ProductAttributeValueExtended> ParseProductAttributes() {
```

Comment:
mmm, that's concerning. Would you mind opening an issue so that I can review that there?